crates/subspace-service/src/dsn.rs

```diff
@@ -44,7 +47,10 @@ where
     trace!("Subspace networking starting.");
 
     let networking_config = subspace_networking::Config::<
-        CustomRecordStore<MemoryRecordStorage, MemoryProviderStorage>,
+        CustomRecordStore<
+            LimitedSizeRecordStorageWrapper<MemoryRecordStorage>,
+            MemoryProviderStorage,
+        >,
```

Comment:
Hm... IMHO nodes shouldn't accept puts at all, they are only supposed to serve the pieces they have recently produced themselves. I mean they might in addition to that, but I don't think they need to or that we discussed this before.